### DISCUSSION TITLE
traefik broken wss wwwwwwwwww

### DISCUSSION THREAD

**Discussion by vectorez** [chatwoot/chatwoot#8528] [Self hosted]
### Describe the bug

Describe the bug

Setting FORCE_SSL=true in Chatwood Self-Hosted (Docker) breaks WebSockets

To Reproduce

Setting FORCE_SSL=true in .env file (as described on https://www.chatwoot.com/docs/self-hosted/deployment/docker )
Adding a WebSite Channel and ingerating in
WebChat is not working, because wss:// is getting redirected with Status 301 to https:// and the WebChat does not seem to handle redirects
Browser logs

-- Console:
action_cable.js:241 WebSocket connection to 'wss://<domain.tld>/cable' failed: Error during WebSocket handshake: Unexpected response code: 404

--- Network:

GET /cable HTTP/1.1
Host: <domain.tld>
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:95.0) Gecko/20100101 Firefox/95.0
Accept: /
Accept-Language: de,en-US;q=0.7,en;q=0.3
Accept-Encoding: gzip, deflate, br
Sec-WebSocket-Version: 13
Origin: https://<domain.tld>
Sec-WebSocket-Protocol: actioncable-v1-json, actioncable-unsupported
Sec-WebSocket-Extensions: permessage-deflate
Sec-WebSocket-Key: OVGgLOlWwrbiF/IcmC7D7w==
Connection: keep-alive, Upgrade
Cookie: StickyCookie=63c648c9568b113d
Sec-Fetch-Dest: websocket
Sec-Fetch-Mode: websocket
Sec-Fetch-Site: same-origin
Pragma: no-cache
Cache-Control: no-cache
Upgrade: websocket

HTTP/1.1 301 Moved Permanently
Content-Type: text/html
Location: https://<domain.tld>/cable
Date: Thu, 06 Jan 2022 22:02:52 GMT
Transfer-Encoding: chunked

Environment

Self Hosted (Docker) with Traefik v2 as Forward Proxy in a testing environment
Tested with recent Firefox and Chromium.

Everything works fine with
FORCE_SSL=false in env file.

Took me some time, because i thought the issue was caused by traefik. But setting the FORCE_SSL to true breaks my WebChat reproducibly.

### To Reproduce

111111111111111

### Environment

Docker

### Cloud Provider

None

### Platform

Browser

**Comment by pranavrajs** [comment]
@vectorez This is an issue with how the reverse proxy is setup. If you could share the configuration, I can help. Moving this to discussions as it is not a bug.

**Reply by vectorez** [reply]
I am using https://easypanel.io/ to manage Docker instances, I used this template https://easypanel.io/docs/templates/chatwoot.
Previously, the issue had already been mentioned, see attached reference.

https://github.com/chatwoot/chatwoot/issues/3704

**Reply by si458** [reply]
sorry to tag onto this, ive just experienced the same problem,
after spending the best part of half a day trying to get `/cable` working and not succeeding.
i had set `FORCE_SSL=true` and it wasnt working
BUT set `FORCE_SSL=false` and it starts working!
im also using traefik, i havent however tested any other type of reverse proxy tho as im only setup for traefik
